Kernel/Memory/MemoryManager.cpp

```diff
@@ -725,6 +725,15 @@ ErrorOr<NonnullOwnPtr<Region>> MemoryManager::allocate_contiguous_kernel_region(
     return allocate_kernel_region_with_vmobject(range, move(vmobject), name, access, cacheable);
 }
 
+ErrorOr<NonnullOwnPtr<Memory::Region>> MemoryManager::allocate_dma_buffer(size_t size, StringView name, Memory::Region::Access access, RefPtr<Memory::PhysicalPage>& dma_buffer_page)
```

Comment:
Hmm, this API is a bit strange. What is supposed to happen if I pass a `size` larger than `PAGE_SIZE`?

Reply:
Good catch.
I added two APIs instead of one now: 
```cpp
ErrorOr<NonnullOwnPtr<Memory::Region>> allocate_dma_buffer_page(StringView name, Memory::Region::Access access, RefPtr<Memory::PhysicalPage>& dma_buffer_page);
ErrorOr<NonnullOwnPtr<Memory::Region>> allocate_dma_buffer_pages(size_t size, StringView name, Memory::Region::Access access, NonnullRefPtrVector<Memory::PhysicalPage>& dma_buffer_pages);
```
I hope the API makes sense now.